### DISCUSSION TITLE
Disable binary compression on build for script injection doesn't work with x86 BIOS ISO

### DISCUSSION THREAD

**Discussion by alexanderturner** [ipxe/ipxe#769] [Q&A]
We have a unique use-case in which we are generating custom ipxe builds for various environments and have a requirement to generate them particularly quickly. To do this, we inject a script with a known number of a single character and perform a sed on the ipxe binary. This process works for tftp booting and EFI ISO. It doesn't work however with BIOS ISO issuing the error:

```
Loading ipxe.lkrn... failed: Invalid argument
```

We achieve this by disabling compression tools in Makefile.housekeeping and replacing `$$(BIN)/%.$(1) : $$(BIN)/%.$(1).zbin` with `$$(BIN)/%.$(1) : $$(BIN)/%.$(1).bin`

This seems to work with other builds but not with BIOS. Can anyone point me in the right direction? How can binary compression be disabled

**Comment by NiKiZe** [comment]
You should not disable compression. However if you are using the iso you could have the script on the iso filesystem instead.

iPXE should have build settings for this that adds the initrd to isolinux.cfg. Still looking for what exactly you need.

**Reply by alexanderturner** [reply]
Thanks @NiKiZe - unfortunately it's not just the ISO, we're also producing builds for PXE boot as well. We have a requirement to switch out the embed script on download (i.e. when the previous link in the chainloader calls it) and we don't have time to wait for the image to recompile (and can't pass initrd args). Ideally, we disable compression and swap out the 'template' we compile iPXE with.

**Reply by NiKiZe** [reply]
The supported way is documented in https://github.com/ipxe/ipxe/commit/a3f1e8fb6707811e6eb90e339d7ebe813fd89a63

**Reply by NiKiZe** [reply]
Is this PXE environment or how is it called since you say no control over initrd?

**Reply by alexanderturner** [reply]
Thnks for your quick replies. We'd like a single build process to ship a hybrid iso as well as a TFTP bootable (chainloadable) image. Our application presents custom data to the user to boot machines. Presently we try to disable compression on the image such that we can SED the resulting file and replace a template which we can do much faster than a build.

If we wanted to (even in an unsupported way) disable compression in the PC BIOS bootloader, where is this done?
